broker/consumer.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python3.7
+# -*- coding: UTF-8 -*-
+
+"""The ``consumer`` module handles the ingestion of a Kafka alert stream into
+Google Cloud Storage (GCS) and defines default connection settings for
+different Kafka servers.
+
+Usage Example
+-------------
+
+.. code-block:: python
+   :linenos:
+
+   from broker.consumer import GCSKafkaConsumer, DEFAULT_ZTF_CONFIG
+
+   # Define connection configuration using default values as a starting point
+   config = DEFAULT_ZTF_CONFIG.copy()
+   config['sasl.kerberos.keytab'] = '<Path to authentication file>'
+   config['sasl.kerberos.principal'] = '<>'
+   print(config)
+
+   # Create a consumer
+   c = GCSKafkaConsumer(
+       kafka_config=config,
+       bucket_name='my-GCS-bucket-name',
+       kafka_topic='ztf_20200301_programid1',
+       pubsub_topic='my-GCS-PubSub-name',
+       debug=True  # Use debug to run without updating your kafka offset
```

Comment:
Naively following these instructions and did not create the bucket first. Got an error on this line `debug=True`:

`BadRequest: 400 GET https://www.googleapis.com/storage/v1/b/my-GCS-bucket-name-Troys-test?projection=noAcl: Invalid bucket name: 'my-GCS-bucket-name-Troys-test'`

Then it kept trying to connect, and returning errors. `ctrl-c` had no effect and I had to shut down my terminal.

I suggest we either add the line of code that creates the bucket (not sure what this is) or just a note saying the bucket needs to exist.

Reply:
The problem forcing me to shut down the terminal was actually caused by not setting `os.environ['KRB5_CONFIG'] = './krb5.conf'` as detailed in the authentication notebook. This module probably assumes one is already authenticated, and in that case the error returned about the bucket not existing is self-explanatory and not fatal, so the code is probably fine the way it is.